### DISCUSSION TITLE
Allow Projects to create Custom Proposal Templates

### DISCUSSION THREAD

**Discussion by stevedefazio** [opensquare-network/collaboration#958] [Ideas]
It would be great if projects could create Custom Proposal Templates.

For example, let's say I'm a DEX and I want to present users with a "Listing Proposal". The team could go into Subsquare, and create a Template:

Period is set to 1 week
Snapshot set to a random time in the prior week
Answers preset to Yes and No
Across the top of the page will be a set of instructions/link to our documentation about the process
The template is available to All Users*

Proposals using this template will be flagged as a "Listing Proposal" when viewed on the main page
Teams can decide which templates are presented to Users

'*' Some proposal templates might be restricted to holders of a certain amount of a token, or might be restricted to whitelisted wallet addresses (for when the team wants to poll the community)

**Comment by nud3l** [comment]
That would be fantastic!

We would love to have the option to create custom templates and just listing some use cases below:
- Allow users to add new pairs to the AMM
- Allow users to create new markets in the lending protocol
- Allow users to add new collateral assets to the bridge
- Adjust risk parameters in the bridge and lending markets

**Reply by robyprastyo** [reply]
Riet

**Comment by wliyongfeng** [comment]
Thanks for the ideas. I agree that template is an useful feature, and several projects has proposed it to us. We will extract common features about template considering different teams have different requests about template. I think common features include:
1. Template title & content.
2. Choices.

Features not sure:
1. Random snapshot height.
2. Threshold for voting or creating proposal.
3. Different space customizations.

About development

- It's better if we implement admins feature before template, so admins can create templates and we don't have to hard code templates first in this repo. We have implemented creating a space feature, check it [here](https://review.opensquare.io/space/new). Admins support won't be very long.
- I can't guarantee the priority of this feature since our capacity is limited and there are lots of pending tasks with subsquare, dotreasury, etc.
- Currently we have no plans to charge fees for this feature to space projects considering it's a common feature. We may seek fund from dotsama treasury in our next proposal to support this development.

**Reply by stevedefazio** [reply]
you absolutely should seek a grant from the treasury. Loads of projects use you guys, you're basically a common good. Please let let us (and all your other users) know when you apply so we can support you

It does make sense to create Admins feature first, since you'd need to be an admin to create a Template

I understand that prioritization is always a challenge.
